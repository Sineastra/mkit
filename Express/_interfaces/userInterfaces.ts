
export interface userInterface {
	name: String,
	hashedPassword: String,
	favourites: string[],
	notes: { movieId: string, note: string }[],
	ratings: { movieId: string, rating: string }[],
}