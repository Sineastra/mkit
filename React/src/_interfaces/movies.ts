export interface movieIntF {
	_id?: string,
	externalId: string,
	poster: string,
	title: string,
	genres: string[],
	runtime: number,
	officialSite: string,
	description: string,
}

export interface movieDetailsForUserIntF {
	rating: number
	notes: string
}