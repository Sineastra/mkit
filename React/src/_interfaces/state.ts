export interface movieInteractionIntF {
	_id: string,
	rating: number,
	privateComment: string,
}
export interface userData {
	_id: string,
	name: string,
	iat?: number,
}
export interface userSliceIntF {
	loggedUser: userData | null
}