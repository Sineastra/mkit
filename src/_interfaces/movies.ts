export interface FavCardMovieIntF {
	imdbId: string,
	poster: string,
	title: string,
	genre: string,
}
