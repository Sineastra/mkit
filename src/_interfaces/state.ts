export interface ratingIntF {
	_id: string,
	rating: number,
	privateComment: string,
}
export interface userData {
	favourites: string[],
	ratings: ratingIntF[]
}