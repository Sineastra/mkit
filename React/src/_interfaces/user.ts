import { movieIntF } from "./movies"


export interface user {
	_id: string,
	name: String,
	hashedPassword: String,
	favourites: movieIntF[],
	notes: { movieId: string, note: string }[],
	ratings: { movieId: string, rating: string }[],
}
