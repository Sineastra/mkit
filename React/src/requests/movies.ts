import { movieIntF } from "../_interfaces/movies"
import { abstractRequest } from "./abstractRequests"



interface moviesRequestsIntF {
	getDetails: (imdbId: string) => Promise<movieIntF>,
	getSearchs: (q: string) => Promise<movieIntF[]>
}
const movieRequests: moviesRequestsIntF = {
	getSearchs: async (query: string) =>
		await abstractRequest({ uri: `/movies${ query }` }),
	getDetails: async (movieId) =>
		await abstractRequest({ uri: `/movies/${ movieId }` }),

}

export { movieRequests }