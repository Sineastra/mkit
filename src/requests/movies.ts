import { FavCardMovieIntF, SearchCardMovieIntF } from "../_interfaces/movies"
import { baseAPIRequest } from "./baseRequest"


interface moviesRequestsIntF {
	getFavs: (ids: string[]) => Promise<FavCardMovieIntF[]>,
	getSearchs: (query: string) => Promise<SearchCardMovieIntF[]>,
}

const movieRequests: moviesRequestsIntF = {
	getFavs: async (ids) => {
		const fetchReq = async (id: string): Promise<FavCardMovieIntF> => {
			const movieData = await baseAPIRequest(`/lookup/shows?imdb=${ id }`)

			return {
				imdbId: movieData.externals.imdb,
				poster: movieData.image.medium,
				title: movieData.name,
				genres: movieData.genres,
			}
		}
		const promises = ids.map(fetchReq)
		const movies: FavCardMovieIntF[] = await Promise.all(promises)

		return movies
	},
	getSearchs: async (query) => {
		const movieData = await baseAPIRequest(`/search/shows?q=${ query }`)
		const movies: SearchCardMovieIntF[] = movieData.map((x: any) => {
			return {
				imdbId: x.externals.imdb,
				poster: x.image.medium,
				title: x.name,
				genres: x.genres,
				runtime: x.runtime,
				officialSite: x.officialSite,
				description: x.summary,
			}
		})

		return movies
	},
}

export { movieRequests }