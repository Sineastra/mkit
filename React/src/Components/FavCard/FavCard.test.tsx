import { movieIntF } from "../../_interfaces/movies"
import renderer from "react-test-renderer"
import { BrowserRouter } from "react-router-dom"
import SingleGridMovie from "./FavCard"


const mockedMovie: movieIntF = {
	externalId: 'a',
	poster: 'b',
	title: 'c',
	genres: ['d', 'e'],
	runtime: 1,
	officialSite: 'b',
	description: 'b',
}

describe("---> Testing /Components/FavCard", () => {
	it("snapshot test", () => {
		const tree = renderer.create(
			<BrowserRouter>
				<SingleGridMovie movie={ mockedMovie }/>
			</BrowserRouter>,
		).toJSON()

		expect(tree).toMatchSnapshot()
	})
})