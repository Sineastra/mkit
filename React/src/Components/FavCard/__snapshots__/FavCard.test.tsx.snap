// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`---> Testing /Components/FavCard snapshot test 1`] = `
<div>
  <a
    className="link"
    href="/details/a"
    onClick={[Function]}
  >
    <img
      alt="poster"
      className="slide"
      src="b"
    />
    <h4>
      c
    </h4>
    <h5>
      d, e
    </h5>
  </a>
</div>
`;
